connect-button/src/test/java/com/ifttt/connect/ui/BaseConnectButtonTest.java

```diff
@@ -139,4 +150,42 @@ public void onError(ErrorResponse errorResponse) {
         assertThat(currentStateRef.get()).isEqualTo(ConnectButtonState.Initial);
         assertThat(errorRef.get()).isNotNull();
     }
+
+    @Test
+    public void testEnabledStateDispatch() throws IOException {
+        AtomicReference<ConnectButtonState> currentStateRef = new AtomicReference<>(ConnectButtonState.Initial);
+
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream("connection_enabled.json");
+        MockWebServer server = new MockWebServer();
+        server.enqueue(new MockResponse().setBody(Okio.buffer(Okio.source(inputStream)).readUtf8()));
+        server.start();
+
+        button.setup("a@b.com",
+            TestUtils.getMockConnectionApiClient(button.getContext(), server),
+            Uri.parse("https://google.com"),
+            credentialsProvider,
+            null
+        );
+
+        Connection connection = loadConnection(getClass().getClassLoader());
+        button.setConnection(connection);
+        button.addButtonStateChangeListener(new ButtonStateChangeListener() {
+            @Override
+            public void onStateChanged(
+                ConnectButtonState currentState, ConnectButtonState previousState, Connection connection
+            ) {
+                currentStateRef.set(currentState);
+            }
+
+            @Override
+            public void onError(ErrorResponse errorResponse) {
+            }
+        });
+
+        button.setConnectResult(new ConnectResult(ConnectResult.NextStep.Complete, "token", null));
+        Shadows.shadowOf(Looper.getMainLooper()).idle();
+
+        assertThat(currentStateRef.get()).isEqualTo(ConnectButtonState.Enabled);
+        server.shutdown();
+    }
```

Comment:
Should we add some logic for status = initial, and an assertion around it before the state change event?

Reply:
Added an assertion for the previous state.